public/app/plugins/datasource/loki/components/monaco-query-field/monaco-completion-provider/CompletionDataProvider.ts

```diff
@@ -0,0 +1,52 @@
+import { HistoryItem } from '@grafana/data';
+import { escapeLabelValueInExactSelector } from 'app/plugins/datasource/prometheus/language_utils';
+
+import LanguageProvider from '../../../LanguageProvider';
+import { LokiQuery } from '../../../types';
+
+import { Label } from './situation';
+
+export class CompletionDataProvider {
+  constructor(private languageProvider: LanguageProvider, private history: Array<HistoryItem<LokiQuery>> = []) {}
+
+  private buildSelector(labels: Label[]): string {
+    const allLabelTexts = labels.map(
+      (label) => `${label.name}${label.op}"${escapeLabelValueInExactSelector(label.value)}"`
+    );
+
+    return `{${allLabelTexts.join(',')}}`;
+  }
+
+  getHistory() {
```

Comment:
I don't think this history works anymore. Not related to this PR, but is always returns just empty array.

Reply:
We have a `history` prop that we pass in `LokiQueryField.tsx`. Can please you confirm that it will always be empty? And maybe add some context to better understand why it will not work, because maybe we should remove it altogether?

Reply:
Sorry, your code is correct, but the history is somehow broken, because I never get history autocomplete suggestions and when I console.log it, it is always empty. So nothing related to this PR, just a note that we should look into history autocomplete part. I'll create separate issue for this.

Reply:
Ohh gotcha. Yes, I understood that part, but had doubts on whether we should remove it or not. Thanks!